src/pools/xaveFxPool/fxPool.ts

```diff
@@ -233,8 +277,12 @@ export class FxPool implements PoolBase<FxPoolPairData> {
         amount: OldBigNumber
     ): OldBigNumber {
         try {
-            return _exactTokenInForTokenOut(amount, poolPairData);
-        } catch {
+            return this._inHigherPrecision(
+                _exactTokenInForTokenOut,
+                amount,
+                poolPairData
+            );
```

Comment:
It seems this one might have to wait until we have SOR converted to fixed-point math as well.

For example, in `_derivativeSpotPriceAfterSwapTokenInForExactTokenOut`, because we still have to use `OldBigNumber` in `_spotPriceAfterSwapTokenInForExactTokenOut`, that means that for both `spotPriceBeforeSwap` and `_spotPriceAfterSwapTokenInForExactTokenOut` we're doing `.div(OLD_ONE_36).decimalPlaces(....)`. I believe this will be easier to fix once we have fixed-point math across SOR. Thoughts?

<img width="1105" alt="Screenshot 2023-06-28 at 11 26 00" src="https://github.com/balancer/balancer-sor/assets/342638/f201fc22-6058-4003-bba9-485a301ac723">